g2o/core/base_constant_edge.hpp

```diff
@@ -0,0 +1,271 @@
+      this->robustKernel()->robustify(error, rho);
+      auto weightedOmega = rho[1] * omega; // this->robustInformation(rho);
+      //std::cout << PVAR(rho.transpose()) << std::endl;
+      //std::cout << PVAR(weightedOmega) << std::endl;
+
+      omega_r *= rho[1];
+      if (fromNotFixed) {
+        {
+          QuadraticFormLock lck(*from);
+
+          from->b().noalias() += A.transpose() * omega_r;
+          from->A().noalias() += A.transpose() * weightedOmega * A;
+        }
+
+        if (toNotFixed ) {
+          if (_hessianRowMajor) // we have to write to the block as transposed
+            hessianTransposed.noalias() += B.transpose() * weightedOmega * A;
+          else
+            hessian.noalias() += A.transpose() * weightedOmega * B;
+        }
+      }
+      if (toNotFixed) {
+        QuadraticFormLock lck(*to);
+
+        to->b().noalias() += B.transpose() * omega_r;
+        to->A().noalias() += B.transpose() * weightedOmega * B;
+      }
+    }
+  }
+};
+
+template <int D, typename E, typename... VertexTypes>
+void BaseConstantEdge<D, E, VertexTypes...>::linearizeOplus(JacobianWorkspace& jacobianWorkspace)
+{
+  linearizeOplus_allocate(jacobianWorkspace, std::make_integer_sequence<int, _nr_of_vertices>());
+  linearizeOplus();
+}
+
+template <int D, typename E, typename... VertexTypes>
+template<int... Ints>
+void BaseConstantEdge<D, E, VertexTypes...>::linearizeOplus_allocate(JacobianWorkspace& jacobianWorkspace, std::integer_sequence<int, Ints...>)
+{
+  ( new (&std::get<Ints>(_jacobianOplus)) JacobianType<D, VertexDimension<Ints>()>(jacobianWorkspace.workspaceForVertex(Ints), D < 0 ? _dimension : D, VertexDimension<Ints>()) , ...);
+}
+
+template <int D, typename E, typename... VertexTypes>
+template<int N>
+void BaseConstantEdge<D, E, VertexTypes...>::linearizeOplusN()
+{
+  auto& jacobianOplus = std::get<N>(_jacobianOplus);
+  auto vertex = vertexXn<N>();
+
+  const number_t delta = cst(1e-9);
+  const number_t scalar = 1 / (2*delta);
+
+  bool iNotFixed = !(vertex->fixed());
+  if (iNotFixed) {
+    QuadraticFormLock lck(*vertex);
+    // estimate the jacobian numerically
+    number_t add_vertex[VertexDimension<N>()] = {};
+
+    // add small step along the unit vector in each dimension
+    for (int d = 0; d < VertexDimension<N>(); ++d) {
+      vertex->push();
+      add_vertex[d] = delta;
+      vertex->oplus(add_vertex);
+      computeError();
+      auto errorBak = this->error();
+      vertex->pop();
+      vertex->push();
+      add_vertex[d] = -delta;
+      vertex->oplus(add_vertex);
+      computeError();
+      errorBak -= this->error();
+      vertex->pop();
+      add_vertex[d] = 0.0;
+
+      jacobianOplus.col(d) = scalar * errorBak;
+    } // end dimension
+  }
+};
+
+template <int D, typename E, typename... VertexTypes>
+template <int... Ints>
+void BaseConstantEdge<D, E, VertexTypes...>::linearizeOplusNs(std::integer_sequence<int, Ints...>)
+{
+  ( void(linearizeOplusN<Ints>()), ...);
```

Comment:
You could just require C++11 if instead of using this you just used regular expansion using this trick:

int unused[] = { (linearizeOPlusN<Ints>(), 0)… };
(void)unused;

Reply:
Would that work for all the fold expressions I used?
If so, then that would be great, because it would mean we can drop the dependency on c++17.

Reply:
Yeah pretty much all your folds can be replaced, like for example:

Even the ones that use the results can be changed like so:
` 
bool fixed[] = {
    vertexXn<Ints>()->fixed()...
};
return std::all_of(std::begin(fixed), std::end(fixed), [](bool value){ return value; });
`

Reply:
I followed your advise and changed all code to make it c++11 compliant.